Sources/ConsoleKit/Utilities/ConsoleLogger.swift

```diff
@@ -116,12 +178,29 @@ extension LoggingSystem {
     ///
     /// - Parameters:
     ///   - console: The console the logger will log the messages to.
-    ///   - level: The minimum level of message that the logger will output. This defaults to `.debug`, the lowest level.
-    ///   - metadata: Extra metadata to log with the message. This defaults to an empty dictionary.
-    public static func bootstrap(console: Console, level: Logger.Level = .info, metadata: Logger.Metadata = [:]) {
-        self.bootstrap { label in
-            return ConsoleLogger(label: label, console: console, level: level, metadata: metadata)
-        }
+    ///   - level: The minimum level of message that the logger will output. This defaults to `.info`.
+    ///   - metadata: Extra metadata to log with all messages. This defaults to an empty dictionary.
+    ///   - metadataProvider: The metadata provider to bootstrap the logging system with.
+    public static func bootstrap(console: Console, level: Logger.Level = .info, metadata: Logger.Metadata = [:], metadataProvider: Logger.MetadataProvider? = nil) {
+        self.bootstrap({ (label, metadataProvider) in
+            return ConsoleLogger(label: label, console: console, level: level, metadata: metadata, metadataProvider: metadataProvider)
+        }, metadataProvider: metadataProvider)
+    }
+    
+    /// Bootstraps a `ConsoleFragmentLogger` to the `LoggingSystem`, so that logger will be used in `Logger.init(label:)`.
+    ///
+    ///     LoggingSystem.boostrap(console: console)
+    ///
+    /// - Parameters:
+    ///   - fragment: The logger fragment which will be used to build the logged messages.
+    ///   - console: The console the logger will log the messages to.
+    ///   - level: The minimum level of message that the logger will output. This defaults to `.info`.
+    ///   - metadata: Extra metadata to log with all messages. This defaults to an empty dictionary.
+    ///   - metadataProvider: The metadata provider to bootstrap the logging system with.
+    public static func bootstrap<T: LoggerFragment>(fragment: T, console: Console, level: Logger.Level = .info, metadata: Logger.Metadata = [:], metadataProvider: Logger.MetadataProvider? = nil) {
+        self.bootstrap({ (label, metadataProvider) in
+            return ConsoleFragmentLogger(fragment: fragment, label: label, console: console, level: level, metadata: metadata, metadataProvider: metadataProvider)
+        }, metadataProvider: metadataProvider)
     }
```

Comment:
Since we're bumping Vapor to require Swift 5.7 for `Sendable` support, let's do the same here (change the `swift-tools-version` in `Package.swift` from 5.6 to 5.7, the CI will adjust automatically). That confers the bonus of the ability to write this method like so:
```suggestion
    public static func bootstrap(
        fragment: some LoggerFragment,
        console: any Console,
        level: Logger.Level = .info,
        metadata: Logger.Metadata = [:],
        metadataProvider: Logger.MetadataProvider? = nil
    ) {
        self.bootstrap({
            ConsoleFragmentLogger(
                fragment: fragment,
                label: $0,
                console: console,
                level: level,
                metadata: metadata,
                metadataProvider: $1
            )
        }, metadataProvider: metadataProvider)
    }
```
Which, while I'm coming up with more annoying things for you to do, reminds me - add `.enableUpcomingFeature("ExistentialAny")` to the `swiftSettings` for the various targets in `Package@swift-5.9.swift` and fix all the resulting error (which will consist of adding `any` in a whole pile of places). Don't worry, the `any` syntax was introduced in 5.6, it'll build without version conditionals; the feature flag just didn't come along until 5.8.

I would recommend also adding these (which combined with the rest would give you the same set I'm using for the updated MySQLNIO), but I'll let that be optional for now:
```swift
                .enableUpcomingFeature("ForwardTrailingClosures"),
                .enableUpcomingFeature("ConciseMagicFile"),
```